### ISSUE TITLE
dcnm_inventory state overridden is not idempotent

### ISSUE THREAD

**Issue by gve-vse-tim** [CiscoDevNet/ansible-dcnm#87]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Ansible Version and collection version

Ansible: 2.10.4

Ansible Collection: cisco.dcnm = 1.2.0

Python modules
- ansible-base 2.10.4
- ansible 2.10.5

### DCNM version

DCNM LAN Fabric: 11.5.1

### Affected module(s)

* dcnm_inventory

### Ansible Playbook

```yaml
- hosts: dcnm
  gather_facts: false
  connection: ansible.netcommon.httpapi

  collections:
    - cisco.dcnm

  vars:
    ansible_command_timeout: 1800
    ansible_connect_timeout: 1800
    switch_username: USERNAME
    switch_password: PASSWORD
    site1_vxlan: VXLAN-Site1
    site1_bgw1: 1.1.3.1
    site1_bgw2: 1.1.3.2
    site2_vxlan: VXLAN-Site2
    site2_bgw1: 2.1.3.1
    site2_bgw2: 2.1.3.2

  tasks:
    - name: Add Border Gateways to "{{ site1_vxlan }}"
      dcnm_inventory:
        fabric: "{{ site1_vxlan }}"
        state: overridden
        config:
          - seed_ip: "{{ site1_bgw1 }}"
            max_hops: 0
            preserve_config: false
            role: border_gateway
            auth_proto: MD5
            user_name: "{{ switch_username }}"
            password: "{{ switch_password }}"
          - seed_ip: "{{ site1_bgw2 }}"
            max_hops: 0
            preserve_config: false
            role: border_gateway
            auth_proto: MD5
            user_name: "{{ switch_username }}"
            password: "{{ switch_password }}"
```

### Debug Output

```bash
# First run 
% ansible-playbook 01-vxlan_fabric_switches.yaml 

PLAY [dcnm] **********************************************************************************************************

TASK [Add Border Gateways to "VXLAN-Site1"] **************************************************************************
[WARNING]: Adding switches to a VXLAN fabric can take a while.  Please be patient...
changed: [dcpod_dcnm]

PLAY RECAP ***********************************************************************************************************
dcpod_dcnm                 : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

# Second run
(ansible-2.10) timmil@TIMMIL-M-842A ansible-dcnm-examples % ansible-playbook 01-vxlan_fabric_switches.yaml 

PLAY [dcnm] **********************************************************************************************************

TASK [Add Border Gateways to "VXLAN-Site1"] **************************************************************************
[WARNING]: Adding switches to a VXLAN fabric can take a while.  Please be patient...
changed: [dcpod_dcnm]

PLAY RECAP ***********************************************************************************************************
dcpod_dcnm                 : ok=0    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

### Expected Behavior

- First run should have completed successfully, taking a great deal of time.
- Second run should have completed with a task status of "ok".
- Second run should have completed much more QUICKLY.
- 
### Actual Behavior

First run does take a LONG time to complete but completes successfully.  Even in scenarios when the switches already existed in the specified roles.

However, subsequent runs take an equally long amount of time and do not simply return "ok".

The documentation clearly says:

```bash
If the switch exists, properties that need to be modified and can be modified will be modified
```

However, each and every run of dcnm_inventory where the state is **overridden** results in the switches being **REMOVED** from the fabric and re-discovered, forcing migration mode, a save/deploy cycle, etc. **despite** the attributes not having changed at all.

Maybe that is somehow (in a non-intuitive way) what "overridden" was intended to mean but that's clearly not what the documentation states.

### Steps to Reproduce

Simply run **ansible-playbook 01-vxlan_fabric_switches.yaml** (YAML copied above) multiple times.

### References

Definitely related to #86 in that the idempotency (do nothing if nothing has changed) is missing for dcnm_inventory.  Perhaps an overridden "state" is needed to behave as it currently does.  However, we need a state (certainly **merge**, if not more) where, if the Ansible doesn't change and the switch in DCNM hasn't been changed, nothing is applied to DCNM.

**Comment by mikewiebe** [comment]
@gve-vse-tim You are correct that state `overridden` should not behave in this manner (bug) and the task should complete more quickly if the switches have already been added and are in the correct state.  We have been planning to revisit this module to speed it up using the bulk add api's instead of adding them individually so we will investigate and determine the best process for improving the performance and predictability of this module.

**Comment by mikewiebe** [comment]
@gve-vse-tim We just merged the fix for this issue (https://github.com/CiscoDevNet/ansible-dcnm/pull/89).  Can you re-try in  your setup and let us know if it resolves the problem you are seeing?

**Comment by gve-vse-tim** [comment]
@mikewiebe Tested it this past week and all seems fine.  Thanks for the fix!

**Comment by mikewiebe** [comment]
Fix now available in `2.0.0` release.
